services/edit/cache.js

```diff
@@ -100,13 +100,39 @@ function getSchema(uri) {
 }
 
 /**
+ * save component, get saved data back
  * @param {object} data
  * @returns {Promise}
  */
 function saveThrough(data) {
   var uri = data[references.referenceProperty];
 
   return removeExtras(uri, data).then(function (data) {
+    return db.save(uri, data);
+  }).then(function (result) {
+    // only clear cache if save is successful
+    exports.getData.cache = new _.memoize.Cache();
+    exports.getDataOnly.cache = new _.memoize.Cache();
+
+    // remember new value (it is returned when save is successful; common with REST implementations)
+    result[references.referenceProperty] = uri;
+    control.setReadOnly(result);
+    exports.getDataOnly.cache.set(uri, result);
+
+    // cache version with schema, return version with schema
+    return exports.getData(uri);
+  });
+}
+
+/**
+ * save component, get html back
+ * @param {object} data
+ * @returns {Promise}
+ */
+function saveForHTML(data) {
```

Comment:
I brought `saveThrough` back (since it's still useful), and moved the "save but gimme html plz" logic to a new function, `saveForHTML`